lib/request.js

```diff
@@ -319,7 +319,14 @@ module.exports = {
     var type = this.get('Content-Type');
     if (!type) return '';
 
-    return contentType.parse(type).parameters.charset || '';
+    try {
+      type = contentType.parse(type);
+    } catch (e) {
+      if (e instanceof TypeError) return '';
+      throw e
```

Comment:
While I agree that `contentType.parse` only throws a `TypeError` right now, there could also be a concern that another error could be thrown in the future (for example: a _SyntaxError_ introduced by a bug).

Koa has a declared dependency of: `"content-type": "^1.0.0",`. Adding a new error type could be conceivable without a major bump which would, in turn, then break this implementation.

It's unlikely but not outside the realm of possibility :)